### PR TITLE
Disable session caching

### LINKED ISSUE #105: SSL checks short-circuit to 'passed' after first check on same host
Because of session caching in the requests library, all stepCI SSL checks to the same host will always pass after the first one. This is because the call to `getPeerCertificate()` fails after the first request to a host, and stepCI then doesn't check the cert at all and falls back to default behaviour. Here is a minimal reproduction:

```js
import { run } from '@stepci/runner'
let workflow = {
  version: "1.1",
  name: "workflow_one",
  tests: {
    example: {
      steps: [{
        name: "step_1",
        http: {
          url: "https://www.google.com",
          method: "GET",
          check: {
            ssl: {
              valid: true,
              signed: true,
              daysUntilExpiration: [{
                  gte: 2000
              }]
            }
          }
        }
      }]
    }
  }
}

run(workflow).then(console.log)
await new Promise(r => setTimeout(r, 1000));
run(workflow).then(console.log)
```

If you run this the first test will fail (as it should, google's cert shouldn't be valid for 2000 days), but the same test will pass the second time around.

A solution to this is to supply got with an https agent with `maxCachedSessions` set to zero as seen here:
https://github.com/hamishfagg/runner/blob/bf73db7e17fa2035f9f829fb45e93c8a150df1e6/src/index.ts#L599-L622

With this the cert is always available, and can be checked for each request.

### PR DESCRIPTION
**Pull request by hamishfagg**
Makes sure that an SSL cert for the host is available for every request.

Fixes #105

### CODE REVIEW

**Comment by mishushakov** [comment]
That was quick 😄
Lemme look, 1s

File: src/index.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ import yaml from 'js-yaml'
 import Ajv from 'ajv'
 import addFormats from 'ajv-formats'
 import { PeerCertificate, TLSSocket } from 'node:tls'
+import { Agent } from 'node:https';
```
Comment (mishushakov) [inline src/index.ts:19]: remove semicolon (sorry bout that, we need to add linting)

File: src/index.ts [hunk 2]
```diff
@@ -595,7 +596,13 @@ async function runTest (id: string, test: Test, schemaValidator: Ajv, options?:
 
           // Make a request
           let sslCertificate: PeerCertificate | undefined
+          let agent = new Agent({
```
Comment (mishushakov) [inline src/index.ts:599]: make it `const`

File: src/index.ts [hunk 3]
```diff
@@ -595,7 +596,13 @@ async function runTest (id: string, test: Test, schemaValidator: Ajv, options?:
...
+            maxCachedSessions: 0
+          });
```
Comment (mishushakov) [inline src/index.ts:601]: remove semicolon (sorry bout that, we need to add linting)

**Review by mishushakov — CHANGES_REQUESTED** [review]

**Comment by hamishfagg** [comment]
Should have addressed those comments now :)

**Pull request merged**